staging/src/k8s.io/component-base/version/version.go

```diff
@@ -137,22 +137,14 @@ func (m *effectiveVersion) Validate() []error {
 	minCompatibilityVersion := m.minCompatibilityVersion.Load()
 
 	// emulationVersion can only be 1.{binaryMinor-3}...1.{binaryMinor}
-	// with a floor of 1.31.
 	maxEmuVer := binaryVersion
 	minEmuVer := binaryVersion.SubtractMinor(3)
-	if minEmuVer.LessThan(floorVersion) {
-		minEmuVer = floorVersion
-	}
 	if emulationVersion.GreaterThan(maxEmuVer) || emulationVersion.LessThan(minEmuVer) {
 		errs = append(errs, fmt.Errorf("emulation version %s is not between [%s, %s]", emulationVersion.String(), minEmuVer.String(), maxEmuVer.String()))
 	}
 	// minCompatibilityVersion can only be 1.{binaryMinor-3} to 1.{binaryMinor-1}
-	// with a floor of 1.31.
 	maxCompVer := binaryVersion.SubtractMinor(1)
```

Comment:
actually, the maxCompVer should be emu version, because you want to be compatible with yourself.

Reply:
Fixed